extras/python/audio2mozzi.py

```diff
@@ -0,0 +1,167 @@
+        num_input_values = int(
+            input_path.stat().st_size / (args.input_bits / 8)
+        )  # Adjust for number format (table at top of https://docs.python.org/3/library/array.html)
+
+        array_type = ""
+        if args.input_bits == 8:
+            array_type = "b"
+        elif args.input_bits == 16:
+            array_type = "h"
+        elif args.input_bits == 32:
+            array_type = "f" if args.input_encoding == "float" else "i"
+
+        valuesfromfile = array.array(array_type)
+        try:
+            valuesfromfile.fromfile(fin, num_input_values)
+        except EOFError:
+            pass
+        in_values = valuesfromfile.tolist()
+
+        tablename = (
+            args.table_name
+            if args.table_name is not None
+            else output_path.stem.replace("-", "_").upper()
+        )
+
+        fout.write(f"#ifndef {tablename}_H_\n")
+        fout.write(f"#define {tablename}_H_\n\n")
+        fout.write("#include <Arduino.h>\n")
+        fout.write('#include "mozzi_pgmspace.h"\n\n')
+        fout.write(f"#define {tablename}_NUM_CELLS {len(in_values)}\n")
+        fout.write(f"#define {tablename}_SAMPLERATE {args.sample_rate}\n\n")
+
+        table = f"CONSTTABLE_STORAGE(int{args.output_bits}_t) {tablename}_DATA [] = {{"
+        max_output_value = 2 << (args.output_bits - 2)  # Halved because signed
+        out_values = []
+        for value in in_values:
+            cnt_33 = 0
+            out_values.append(
+                math.trunc((value * max_output_value) + 0.5)
+                if args.input_encoding == "float"
+                else value
+            )
+            # Mega2560 boards won't upload if there is 33, 33, 33 in the array, so dither the 3rd 33 if there is one
+            if value == 33:
+                cnt_33 += 1
+                if cnt_33 == 3:
+                    out_values.append(random.choice((32, 34)))
+                    cnt_33 = 0
+            else:
+                cnt_33 = 0
+        if args.make_symmetrical:
+            value_to_remove = -max_output_value - 1
+            if value_to_remove in out_values:
+                in_min = value_to_remove
+                in_max = max_output_value
+                out_max = max_output_value
+                out_min = -max_output_value
+                out_values = (((value - in_min) * (out_max - out_min) / (in_max - in_min) + out_min) for value in out_values)
+
+        table += ", ".join(map(str, out_values))
+        table += "};"
+        table = textwrap.fill(table, 80)
+        fout.write(table)
+        fout.write("\n\n")
+        fout.write(f"#endif /* {tablename}_H_ */\n")
+        print(f"wrote {table} to {output_path}")
+
+    return 0
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser(
+        description="Script for converting a raw audio file to a Mozzi table",
+    )
+    parser.add_argument(
+        "-e",
+        "--input-encoding",
+        choices=("float", "int"),
```

Comment:
In float mode, with default number of output bits, some values reach 128 (which do not fit). Maybe you should discard my comment on the make symmetric but subtract -1 to the max output value?

Reply:
uhm, I didn't really understand this. Do you mean that the output of the script reaches 128? That's not expected